databuilder/query_model.py

```diff
@@ -443,6 +452,10 @@ def validate_input_domains(node):
                 f"Attempt to combine unrelated domains:\n{non_patient_domains}"
                 f"\nIn node:\n{node}"
             )
+        if isinstance(node, AggregatedSeries) and len(non_patient_domains) == 0:
+            raise DomainMismatchError(
+                f"Attempt to aggregate one-row-per-patient series\nIn node:\n{node}"
+            )
```

Comment:
This is sufficiently grimace-inducing to motivate something along the lines that you proposed earlier. But totally okay to defer it.